tools/scripts/build_ubuntu_x64_ncnn.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+import sys
+import time
+
+from ubuntu_utils import cmd_result, ensure_base_env, get_job
+
+g_jobs = 2
+
+
+def install_protobuf(dep_dir) -> int:
+    """build and install protobuf.
+
+    Args:
+        wor_dir (_type_): _description_
+
+    Returns:
+        : _description_
+    """
+    print('-' * 10 + 'install protobuf' + '-' * 10)
+
+    os.chdir(dep_dir)
+    if not os.path.exists('protobuf-3.20.0'):
+        os.system(
+            'wget https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protobuf-cpp-3.20.0.tar.gz'  # noqa: E501
+        )
+        os.system('tar xvf protobuf-cpp-3.20.0.tar.gz')
+
+    os.chdir(os.path.join(dep_dir, 'protobuf-3.20.0'))
+
+    install_dir = os.path.join(dep_dir, 'pbinstall')
+    os.system('./configure --prefix={}'.format(install_dir))
+    os.system('make -j {} && make install'.format(g_jobs))
+    protoc = os.path.join(dep_dir, 'pbinstall', 'bin', 'protoc')
+
+    print('protoc \t:{}'.format(cmd_result('{} --version'.format(protoc))))
+    return 0
+
+
+def install_pyncnn(dep_dir):
+    print('-' * 10 + 'build and install pyncnn' + '-' * 10)
+    time.sleep(2)
+
+    # generate unzip and build dir
+    os.chdir(dep_dir)
+
+    # git clone
+    if not os.path.exists('ncnn'):
+        os.system(
+            'git clone https://github.com/tencent/ncnn && cd ncnn && git checkout 20220729'  # noqa: E501
```

Comment:
could clone  a tag branch with only one commit history to speed up downloading time.
```
git clone --depth 1 --branch 20220729  https://github.com/tencent/ncnn
```

Reply:
done.